tests/TransactionTest.php

```diff
@@ -0,0 +1,226 @@
+<?php
+declare(strict_types=1);
+
+use Illuminate\Support\Facades\DB;
+
+class TransactionTest extends TestCase
+{
+    protected $insertData = ['name' => 'klinson', 'age' => 20, 'title' => 'admin'];
+    protected $originData = ['name' => 'users', 'age' => 20, 'title' => 'user'];
+    protected $connection = 'mongodb_repl';
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        User::on($this->connection)->truncate();
+        User::on($this->connection)->create($this->originData);
+    }
+
+    protected function getEnvironmentSetUp($app)
+    {
+        if (version_compare(env('MONGO_VERSION'), '4', '<')) {
+            $this->markTestSkipped('MongoDB with version below 4 is not supported for transactions');
+        }
+
+        $config = require 'config/database.php';
+
+        $app['config']->set('database.connections.'.$this->connection, $config['connections'][$this->connection]);
+        $app['config']->set('database.default', $this->connection);
+    }
+
+    public function tearDown(): void
+    {
+        User::on($this->connection)->truncate();
+
+        parent::tearDown();
+    }
+
+    public function testCreate()
+    {
+        /** rollback test */
+        DB::beginTransaction();
+        $user = User::on($this->connection)->create($this->insertData);
+        DB::rollBack();
+        $this->assertInstanceOf(\Jenssegers\Mongodb\Eloquent\Model::class, $user);
+        $this->assertTrue($user->exists);
```

Comment:
Does the `exists` property have special meaning in Laravel? Is it supposed to indicate whether a model instance is stored in the database, or what is its purpose? I'm asking because the user doesn't actually exist in the database due to the rollback, so the value indicated here may be incorrect.

Reply:
But I think here we don't need to use `find` as it return Model instance. We can use the `exists()` method provided by QueryBuilder to check the existence in the database. It returns boolean value. That is enough to make sure that rollback works.